job.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+const fs = require('fs-extra');
+const _ = require('lodash');
+const DummyLogger = require('sq-logger/dummy-logger');
+const Duration = require('sq-toolkit/duration');
+
+const template = require('./lib/utils/template');
+const BigQueryError = require('./error');
+const BigQueryFactory = require('./factory');
+
+const COST_THRESHOLD_IN_GB = 100;
+
+class BigQueryJob {
+
+    /**
+     * @param {Object}    opts
+     * @param {String=}   opts.name
+     * @param {String}    opts.sqlFilename
+     * @param {Number=}   [opts.costThresholdInGB = 100]
+     *
+     * @param {BigQuery=} opts.bigQuery
+     * @param {WinstonLogger|DummyLogger=} opts.logger
+     */
+    constructor(opts) {
+        this.name = opts.name || null;
+        this.sqlFilename = opts.sqlFilename || null;
+        this.costThresholdInGB = opts.costThresholdInGB || COST_THRESHOLD_IN_GB;
```

Comment:
Compare with the == null here, because this way the opts.costThresholdInGB can not be zero.
this.costThresholdInGB = opts.costThresholdInGB != null ?  opts.costThresholdInGB : COST_THRESHOLD_IN_GB;